File: DarkOrbit 10.0/Net/netty/commands/KillScreenOptionModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class KillScreenOptionModule
    {
        public const short ID = 31092;

        public KillScreenOptionTypeModule repairType;
        public PriceModule price;
        public bool affordableForPlayer;
        public int cooldownTime;
        public MessageLocalizedWildcardCommand descriptionKey;
        public MessageLocalizedWildcardCommand descriptionKeyAddon;
        public MessageLocalizedWildcardCommand toolTipKey;
        public MessageLocalizedWildcardCommand buttonKey;

        public KillScreenOptionModule(KillScreenOptionTypeModule pRepairType, PriceModule pPrice,
                                  bool pAffordableForPlayer, int pCooldownTime,
                                  MessageLocalizedWildcardCommand pDescriptionKey,
                                  MessageLocalizedWildcardCommand pDescriptionKeyAddon,
                                  MessageLocalizedWildcardCommand pToolTipKey,
                                  MessageLocalizedWildcardCommand pButtonKey)
        {
            this.repairType = pRepairType;
            this.price = pPrice;
            this.affordableForPlayer = pAffordableForPlayer;
            this.cooldownTime = pCooldownTime;
            this.descriptionKey = pDescriptionKey;
            this.descriptionKeyAddon = pDescriptionKeyAddon;
            this.toolTipKey = pToolTipKey;
            this.buttonKey = pButtonKey;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.write(descriptionKeyAddon.write());
            param1.write(buttonKey.write());
            param1.write(toolTipKey.write());
            param1.write(price.write());
            param1.write(repairType.write());
            param1.writeBoolean(affordableForPlayer);
            param1.writeShort(-16569);
            param1.writeInt(cooldownTime >> 14 | cooldownTime << 18);
            param1.write(descriptionKey.write());
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/KillScreenOptionModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

/**
 Created by Shock & Pedro on 30-03-2015.
 */
public class KillScreenOptionModule
        implements ServerCommand {

    public static int ID = 30589;

    public KillScreenOptionTypeModule      mRepairType;
    public PriceModule                     mPrice;
    public boolean                         mAffordableForPlayer;
    public int                             mCooldownTime;
    public MessageLocalizedWildcardCommand mDescriptionKey;
    public MessageLocalizedWildcardCommand mDescriptionKeyAddon;
    public MessageLocalizedWildcardCommand mToolTipKey;
    public MessageLocalizedWildcardCommand mButtonKey;

    public KillScreenOptionModule(KillScreenOptionTypeModule pRepairType, PriceModule pPrice,
                                  boolean pAffordableForPlayer, int pCooldownTime,
                                  MessageLocalizedWildcardCommand pDescriptionKey,
                                  MessageLocalizedWildcardCommand pDescriptionKeyAddon,
                                  MessageLocalizedWildcardCommand pToolTipKey,
                                  MessageLocalizedWildcardCommand pButtonKey) {
        this.mRepairType = pRepairType;
        this.mPrice = pPrice;
        this.mAffordableForPlayer = pAffordableForPlayer;
        this.mCooldownTime = pCooldownTime;
        this.mDescriptionKey = pDescriptionKey;
        this.mDescriptionKeyAddon = pDescriptionKeyAddon;
        this.mToolTipKey = pToolTipKey;
        this.mButtonKey = pButtonKey;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeShort(-22445);
            param1.writeInt(this.mCooldownTime >>> 7 | this.mCooldownTime << 25);
            this.mPrice.write(param1);
            this.mToolTipKey.write(param1);//
            this.mDescriptionKeyAddon.write(param1);//
            this.mRepairType.write(param1);
            this.mButtonKey.write(param1);//
            this.mDescriptionKey.write(param1);//
            param1.writeShort(30426);
            param1.writeBoolean(this.mAffordableForPlayer);
        } catch (IOException e) {
        }
    }
}
